spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/builder/ExportGenerator.java

```diff
@@ -0,0 +1,302 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.bootstrap.data.builder;
+
+import org.candlepin.dto.api.client.v1.BrandingDTO;
+import org.candlepin.dto.api.client.v1.CdnDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.ContentDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.dto.api.client.v1.ReleaseVerDTO;
+import org.candlepin.dto.api.client.v1.RoleDTO;
+import org.candlepin.dto.api.client.v1.UserDTO;
+import org.candlepin.invoker.client.ApiException;
+import org.candlepin.resource.client.v1.CdnApi;
+import org.candlepin.resource.client.v1.OwnerContentApi;
+import org.candlepin.resource.client.v1.OwnerProductApi;
+import org.candlepin.resource.client.v1.RolesApi;
+import org.candlepin.resource.client.v1.UsersApi;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.api.ConsumerClient;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+
+import java.io.File;
+import java.time.OffsetDateTime;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ExportGenerator implements AutoCloseable {
```

Comment:
During the exporter spec migration (https://github.com/candlepin/candlepin/pull/3509) after discussion it was decided to not create an 'exporter' abstraction/infrastructure, because it is unnecessarily creating data (products, pools, etc.) all at once, while each test method only needs a minimal amount. And that the export process was primarily slow just because of that creating of a massive amount of data. Instead, each test should create the test data it needed, and then do an export with just that data

Reply:
Done. Tests got significantly slower, but the change got rid of lifetime changes so it might be a worthy tradeoff.

Reply:
I am keeping it as a separate commit for now so that they can be compared.